src/main/java/org/springframework/cloud/deployer/spi/kubernetes/DefaultContainerFactory.java

```diff
@@ -97,9 +97,16 @@ public Container create(ContainerConfiguration containerConfiguration) {
 
 		List<String> appArgs = new ArrayList<>();
 
+		Map<String, String> appAdminCredentials = new HashMap<>();
+		properties.getAppAdmin().addCredentialsToAppEnvironmentAsProperties(appAdminCredentials);
+
 		switch (entryPointStyle) {
 		case exec:
 			appArgs = createCommandArgs(request);
+			List<String> finalAppArgs = appArgs;
+			appAdminCredentials.forEach((k, v) -> finalAppArgs.add(String.format("--%s=%s", k,v)));
```

Comment:
Do these creds end up in the log when container is run?

Reply:
Not the log, but they are visible using `kubectl get pod` or describe for the app container. Any user who has access to the namespace can see them. Not wide open, but not ideal. I'll look into it further.

Reply:
Thanks for adding https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/issues/483